src/components/MoveModal.vue

```diff
@@ -41,18 +45,20 @@ export default {
 			this.moving = true
 
 			try {
-				const envelopeIds = this.envelopes
-					.filter((envelope) => envelope.mailboxId !== this.destMailboxId)
-					.map((envelope) => envelope.databaseId)
+				const envelopes = this.envelopes
+					.filter(envelope => envelope.mailboxId !== this.destMailboxId)
 
-				if (envelopeIds.length === 0) {
+				if (envelopes.length === 0) {
 					return
 				}
 
-				await Promise.all(envelopeIds.map((id) => this.$store.dispatch('moveMessage', {
-					id,
-					destMailboxId: this.destMailboxId,
-				})))
+				for (const envelope of envelopes) {
```

Comment:
The change `Promise.all` to `for...of` is not necessary for the threading improvements. `Promise.all` will dispatch a moveMessage / moveThread action for each envelope passed to the MoveModal. Ideally would be a "loading" state to signal that mail is working on something.